src/components/SettingsForm.vue

```diff
@@ -100,18 +100,44 @@
 				type="text"
 				placeholder="quota">
 		</p>
+		<p>
+			<label for="mapping-quota">{{ t('user_oidc', 'Groups mapping') }}</label>
+			<input id="mapping-quota"
+				v-model="localProvider.settings.mappingGroups"
+				type="text"
+				placeholder="groups"
+				:disabled="!localProvider.settings.groupProvisioning">
+		</p>
 		<NcCheckboxRadioSwitch :checked.sync="localProvider.settings.uniqueUid" wrapper-element="div">
 			{{ t('user_oidc', 'Use unique user id') }}
 		</NcCheckboxRadioSwitch>
 		<p class="settings-hint">
 			{{ t('user_oidc', 'By default every user will get a unique userid that is a hashed value of the provider and user id. This can be turned off but uniqueness of users accross multiple user backends and providers is no longer preserved then.') }}
 		</p>
+		<NcCheckboxRadioSwitch :checked.sync="localProvider.settings.providerBasedId" wrapper-element="div">
+			{{ t('user_oidc', 'Use provider identifier as prefix for ids') }}
+		</NcCheckboxRadioSwitch>
+		<p class="settings-hint">
+			{{ t('user_oidc', 'To keep ids in plain text, but also preserve uniqueness of them across multiple providers, a prefix with the providers name is added.') }}
+		</p>
+		<NcCheckboxRadioSwitch :checked.sync="localProvider.settings.groupProvisioning" wrapper-element="div">
+			{{ t('user_oidc', 'Use group provisioning. Format of the groups claim value should be [{gid: "1", displayName: "group1"}, ...] or ["group1", "group2", ...]') }}
```

Comment:
Could you move "Format of the groups claim value should be..." to the settings hint?